### PR TITLE
[#126] Always return RULE_ENGINE_CONTINUE from exec_rule (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Will test soon.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -353,9 +353,16 @@ namespace irods::plugin::rule_engine::audit_amqp
...
-		irods::error err = _eff_hdlr("unsafe_ms_ctx", &rei);
-		if (!err.ok()) {
-			return err;
+		if (const auto err = _eff_hdlr("unsafe_ms_ctx", &rei); !err.ok()) {
+			// clang-format off
+			log_re::trace({
+				{"rule_engine_plugin", rule_engine_name},
+				{"log_message", "could not get rule execution context (REI)"},
+				{"rule_name", _rn},
+				{"error_result", ret.result()},
```
Comment (trel) [inline src/main.cpp:362]: err.result?

Comment (korydraughn) [reply src/main.cpp:362]: Will tweak.

Comment (korydraughn) [reply src/main.cpp]: Fixed.

File: src/main.cpp [hunk 2]
```diff
@@ -353,9 +353,16 @@ namespace irods::plugin::rule_engine::audit_amqp
...
-		irods::error err = _eff_hdlr("unsafe_ms_ctx", &rei);
-		if (!err.ok()) {
-			return err;
+		if (const auto err = _eff_hdlr("unsafe_ms_ctx", &rei); !err.ok()) {
+			// clang-format off
+			log_re::trace({
+				{"rule_engine_plugin", rule_engine_name},
+				{"log_message", "could not get rule execution context (REI)"},
+				{"rule_name", _rn},
+				{"error_result", err.result()},
```
Comment (alanking) [inline src/main.cpp:362]: Extra comma

File: src/main.cpp [hunk 3]
```diff
@@ -434,18 +441,21 @@ namespace irods::plugin::rule_engine::audit_amqp
...
 			log_exception(e, "Caught exception", {"rule_name", _rn});
-			return ERROR(SYS_INTERNAL_ERR, e.what());
 		}
 		catch (...) {
-			return ERROR(SYS_UNKNOWN_ERROR, "an unknown error occurred");
+			// clang-format off
+			log_re::error({
+				{"rule_engine_plugin", rule_engine_name},
+				{"log_message", "an unknown error occurred"},
+				{"rule_name", _rn},
```
Comment (alanking) [inline src/main.cpp:456]: Extra comma

**Review by alanking — COMMENTED** [review]
Couple of things that apparently do not matter for compilation(?)

Please notify when confirmed working

**Comment by korydraughn** [comment]
Everything is in working order.

File: src/main.cpp (hunk 2 above)
Comment (korydraughn) [reply src/main.cpp:362]: > Couple of things that apparently do not matter for compilation(?)

That's correct. I'm guessing that is a feature of the language. Other languages allow that too.

Will remove.

**Comment by korydraughn** [comment]
I've confirmed that before this PR, the plugin did not allow subsequent plugins to process rules.

File: src/main.cpp (hunk 2 above)
Comment (korydraughn) [reply src/main.cpp]: Removed.

File: src/main.cpp (hunk 3 above)
Comment (korydraughn) [reply src/main.cpp]: Removed.

**Review by alanking — APPROVED** [review]
Yep, let's get it in

**Comment by korydraughn** [comment]
#'d

**Pull request merged**